src/pybind/mgr/volumes/fs/operations/trash.py

```diff
@@ -113,6 +183,50 @@ def delink(self, bname):
         except cephfs.Error as e:
             raise VolumeException(-e.args[0], e.args[1])
 
+    def _get_stats(self, path, num_of_files, num_of_subvols, iter_level):
```

Comment:
Hang on - this is recursively iterating the directory to account for number of files (subvols?). This is going to be expensive.

Reply:
Had a chat with Venky about this one. The idea is to switch to rstats in this case too even though it would have a pretty negative effect on progress bar (since unlink per sec rate is very high, it would mean rstats update would be too slow to for progress bar to move as purges makes progress) since recursive iteration will be too heavy as n op for CephFS.